pkg/skaffold/build/buildpacks/lifecycle.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2019 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package buildpacks
+
+import (
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"io"
+	"path/filepath"
+
+	"github.com/docker/docker/api/types/mount"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest"
+)
+
+func (b *BuildpackBuilder) build(ctx context.Context, out io.Writer, workspace string, artifact *latest.BuildpackArtifact, tag string) (string, error) {
+	// To improve caching, we always build the image with [:latest] tag
```

Comment:
The buildpacks folks mentioned you could walk forward the tags by building with the previous, then retagging with the next, then loop, e.g.:

```
build foo:1
tag foo:1 foo:2
build foo:2
tag foo:2 foo:3
...
```

This might be hard to accomplish and using `:latest` is probably easier, but it's a possibility.